### PR TITLE
Updated MDC plan features

### PR DESCRIPTION
**Pull request by jude-chen**
## Issue number

#000

## Checklist before requesting a review
- [ ] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by elibetito-microsoft** [comment]
@jude-chen could you please resolve the conflicts?

@lonegunmanb - appreciate your review to this PR also. Thanks!

**Comment by lonegunmanb** [comment]
Hi @jude-chen thanks for opening this pr! We're maintaining `module_telemetry.tf` file automatically in our CI pipeline so you can just ignore this file. As your pr added a lot of code, would you please give us a brief on the goal of this pr? We'd like to assess whether we should accept these features in this module. @elibetito-microsoft will review this pr based on the mdc's perspective, and I'll review this pr based on Terraform module's perspective.

**Comment by lonegunmanb** [comment]
I'm closing this pr since no response from the user, please feel free to reopen it @jude-chen

**Comment by jude-chen** [comment]
@lonegunmanb sorry for the delay, I have updated the PR description with the changes I made in this PR. Can you please reopen the PR and review it again? Thanks.

File: module_telemetry.tf
Comment (jude-chen) [inline module_telemetry.tf:1]: Please drop the changes in this file since it is auto-updated.

**Pull request closed without merging**